src/transport_ofi.h

```diff
@@ -299,18 +300,18 @@ void shmem_transport_put_quiet(void)
         success = fi_cntr_read(shmem_transport_ofi_put_cntrfd);
         fail = fi_cntr_readerr(shmem_transport_ofi_put_cntrfd);
 
-        if (success < shmem_transport_ofi_pending_put_counter && fail == 0) {
+        if (success < OPA_load_int(&shmem_transport_ofi_pending_put_counter) && fail == 0) {
             SPINLOCK_BODY();
         }
         else if (fail) {
             struct fi_cq_err_entry e = {0};
             fi_cq_readerr(shmem_transport_ofi_put_nb_cqfd, (void *)&e, 0);
             OFI_CQ_ERROR(shmem_transport_ofi_put_nb_cqfd, &e);
         }
-    } while (success < shmem_transport_ofi_pending_put_counter);
+    } while (success < OPA_load_int(&shmem_transport_ofi_pending_put_counter));
 #else
     int ret = fi_cntr_wait(shmem_transport_ofi_put_cntrfd,
-                           shmem_transport_ofi_pending_put_counter, -1);
+                           OPA_load_int(&shmem_transport_ofi_pending_put_counter), -1);
```

Comment:
I think this needs to be in a loop that repeats the call to ```fi_cntr_wait``` if ```pending_put_counter``` increased during the call.

Reply:
Same comment below with ```get_wait```.